main.rb

```diff
@@ -0,0 +1,186 @@
+# rubocop:disable Metrics/ModuleLength
+# rubocop:disable Metrics/CyclomaticComplexity
+# rubocop:disable Metrics/PerceivedComplexity
+
+module Enumerable
+  def my_each
+    return enum_for(:my_each) unless block_given?
+
+    new_self = self
+    i = 0
+    new_self = new_self.instance_of?(Array) || new_self.instance_of?(Hash) ? flatten : to_a
+    if instance_of?(Array) || instance_of?(Range)
+      new_self.length.times do
+        yield(new_self[i])
+        i += 1
+      end
+    elsif instance_of?(Hash)
+      new_self.length / 2.times do
+        yield(new_self[i], new_self[i + 1])
+        i += 2
+      end
+    end
+    new_self
+  end
+
+  def my_each_with_index
+    return enum_for(:my_each_with_index) unless block_given?
+
+    new_self = self
+    i = 0
+    new_self = new_self.instance_of?(Array) || new_self.instance_of?(Hash) ? flatten : to_a
+    new_self.length.times do
+      yield(new_self[i], i)
+      i += 1
+    end
+    new_self
+  end
+
+  def my_select
+    return enum_for(:my_select) unless block_given?
+
+    i = 0
+    arr = []
+    new_self = self
+    new_self = new_self.instance_of?(Array) || new_self.instance_of?(Hash) ? flatten : to_a
+    new_self.length.times do
+      yield(new_self[i]) == true ? arr = arr.push(new_self[i]) : arr
+      i += 1
+    end
+    arr
+  end
+
+  def my_all?(parameter = nil)
+    i = 0
+    check = 0
+    new_self = self
+    new_self = new_self.instance_of?(Array) || new_self.instance_of?(Hash) ? flatten : to_a
+    while block_given? && i < new_self.length
+      yield(new_self[i]) == true ? check += 1 : check = check
+      i += 1
+    end
+    if parameter.instance_of?(Regexp)
+      new_self.length.times do
+        new_self[i] =~ parameter ? check += 1 : check = check
+        i += 1
+      end
+    else
+      new_self.length.times do
+        new_self[i].class == parameter ? check += 1 : check = check
+        i += 1
+      end
+    end
+    check == new_self.length
+  end
+
+  def my_any?(parameter = nil)
+    i = 0
+    check = 0
+    new_self = self
+    new_self = new_self.instance_of?(Array) || new_self.instance_of?(Hash) ? flatten : to_a
+    while block_given? && i < new_self.length
+      yield(new_self[i]) == true ? check += 1 : check = check
+      i += 1
+    end
+    if parameter.instance_of?(Regexp)
+      new_self.length.times do
+        new_self[i] =~ parameter ? check += 1 : check = check
+        i += 1
+      end
+    else
+      new_self.length.times do
+        new_self[i].class == parameter ? check += 1 : check = check
+        i += 1
+      end
+    end
+    check.positive? ? true : false
+  end
```

Comment:
# my_any
- [ ] `my_any?` when a class is passed as an argument returns true if at least one of the collection is a member of such class::Numeric

     Failure/Error: expect(array.my_any?(Numeric)).to be array.any?(Numeric)

       expected true
            got false
##
- [ ] `my_any?` when a pattern other than Regex or a Class is given returns true if any of the collection matches the pattern

      Failure/Error: expect(words.my_any?('cat')).to be words.any?('cat')

        expected true
             got false
##
# my_any?
- [ ] `my_any?` when no block or argument is given returns true if at least one of the collection is not false or nil

     Failure/Error: expect(true_array.my_any?).to be true_array.any?

       expected true
            got false